### PR TITLE
When json_load_file fails report the error details information

### PR DESCRIPTION
**Pull request by barry-scott**
returned.

This allows for file I/O issues as well as JSON parsing issues to be reported.

### CODE REVIEW

File: src/keys.c [hunk 1]
```diff
@@ -363,6 +363,8 @@ load_keys(const char* jwkdir)
         }
 
         if (strcmp(dot, pattern) == 0) {
+            json_error_t error;
```
Comment (sarroutbi) [inline src/keys.c:366]: Could not this be moved closer to its usage? If no error, this declaration is not required, so I would remove the declaration to block inside if(!json) { ...

File: src/keys.c [hunk 2]
```diff
@@ -371,7 +373,10 @@ load_keys(const char* jwkdir)
             filepath[sizeof(filepath) - 1] = '\0';
             json_auto_t* json = json_load_file(filepath, 0, NULL);
```
Comment (sergio-correia) [inline src/keys.c:374]: Use the `error` here, i.e. `json_load_file(filepath, 0, &error);`

File: src/keys.c [hunk 3]
```diff
@@ -371,7 +373,10 @@ load_keys(const char* jwkdir)
...
             if (!json) {
-                fprintf(stderr, "Invalid JSON file (%s); skipping\n", filepath);
+                enum json_error_code code = json_error_code(&error);
```
Comment (sergio-correia) [inline src/keys.c:376]: This `json_error_code()` was added in jansson 2.11, which is why it's failing in the CI for Centos 7, as it has 2.10. On the other hand, I am not sure it adds much vaulable info in this case. Perhaps you could drop `code` here and in the `fprintf` below, and keep only the text, line, column and position?

File: src/keys.c [hunk 4]
```diff
@@ -371,7 +373,10 @@ load_keys(const char* jwkdir)
...
+                fprintf(stderr, "Cannot load JSON file (%s); skipping\n", filepath);
+                fprintf(stderr, "error code %d text %s line %d col %d pos %d\n",
```
Comment (sergio-correia) [inline src/keys.c:378]: For better readability, could you please add some separator here between the code, text, etc? A comma would suffice, I guess.

**Review by sergio-correia — CHANGES_REQUESTED** [review]

File: src/keys.c (hunk 1 above)
Comment (barry-scott) [reply src/keys.c:366]: You always have a new enough C compiler? I'll move assuming yes.

File: src/keys.c (hunk 2 above)
Comment (barry-scott) [reply src/keys.c:374]: Doh, copied the working patch and missed that.

File: src/keys.c (hunk 3 above)
Comment (barry-scott) [reply src/keys.c:376]: Will drop.

File: src/keys.c (hunk 4 above)
Comment (barry-scott) [reply src/keys.c:378]: Now like this:

`
tangd[18555]: Cannot load JSON file (/var/db/tang/qqq-exc-2022-05-29.jwk); skipping
tangd[18555]: error text unable to open /var/db/tang/qqq-exc-2022-05-29.jwk: Permission denied, line -1, col -1, pos 0
`

**Review by sergio-correia — APPROVED** [review]
Thank you, it looks better now.

**Pull request merged**